lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -5691,13 +5716,64 @@ FIRRTLType RefSubOp::inferReturnType(ValueRange operands,
     return RefType::get(bundleType.getElement(fieldIdx).type.getConstType(
                             bundleType.isConst() ||
                             bundleType.getElement(fieldIdx).type.isConst()),
-                        refType.getForceable());
+                        refType.getForceable(), refType.getLayer());
   }
 
   return emitInferRetTypeError(
       loc, "ref.sub op requires a RefType of vector or bundle base type");
 }
 
+LogicalResult RefCastOp::verify() {
+
+  SymbolRefAttr layer;
+  auto layerBlockOp =
+      dyn_cast<LayerBlockOp>((*this)->getBlock()->getParentOp());
```

Comment:
Can layers have region-creating ops like whens or match statements?

Reply:
They can. This looks like a bug in that case. It should walk backwards until it finds a layerblock (or bottoms out at a module). Good catch.

Reply:
Fixed in 54b7da58e.